controllers/machine_controller.go

```diff
@@ -105,6 +106,18 @@ func (r *MachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr e
 		return ctrl.Result{}, err
 	}
 
+	cluster, err := util.GetClusterByName(ctx, r.Client, m.ObjectMeta.Namespace, m.Spec.ClusterName)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to get cluster %q for machine %q in namespace %q",
+			m.Labels[clusterv1.ClusterLabelName], m.Name, m.Namespace)
```

Comment:
nit: I'd use `m.Spec.ClusterName` instead of `m.Labels[clusterv1.ClusterLabelName]`